mergify_engine/tests/unit/test_clients.py

```diff
@@ -28,40 +28,74 @@
 from mergify_engine.clients import http
 
 
+gh_fake_response = {
+    "id": 12345,
+    "target_type": "User",
+    "permissions": {
+        "checks": "write",
+        "contents": "write",
+        "pull_requests": "write",
+    },
+    "account": {"login": "testing", "id": 12345},
+}
+
+
+def _httpserver_connect(httpserver: httpserver.HTTPServer, endpoint: str) -> None:
```

Comment:
You can go even further by also passing the HTTP client `auth=` to this method, so when the `with AsyncGithubInstallationClient..` and all assert can be moved in this method.